### ISSUE TITLE
Using this tool for another LSUN dataset + model

### ISSUE THREAD

**Issue by doctor-gonzo** [mit-han-lab/anycost-gan#19]
Hello! Thanks for creating this.

I am trying to use this tool with another model (the [LSUN Churches dataset](https://www.tensorflow.org/datasets/catalog/lsun#lsunchurch_outdoor)) with sliders that represent attributes.

As I understand it, these are the steps I need to take to configure this toolset to work with a different dataset + pre-trained network:

- Make sure the format of the LSUN Churches dataset matches as [described for other models](https://github.com/mit-han-lab/anycost-gan#using-pre-trained-models)

- Change the config name as described [here](https://github.com/mit-han-lab/anycost-gan#datasets) to `config_name = ''stylegan2-church-config-f"`, referring to the pre-trained network found [here](https://nvlabs-fi-cdn.nvidia.com/stylegan2/networks/)

- run `models.get_pretrained('attribute-predictor')` as described in the pre-trained-models section of README

- Change the relevant attribute labels in the files that show up during [this search](https://github.com/mit-han-lab/anycost-gan/search?q=Eyeglasses)

I am just wondering if there any obvious steps I am missing to get this working, I am very new to the world of GANs and toolsets. Thank you for your time 😊

**Comment by tonylins** [comment]
Hi, currently we only provide the encoder and editing directions for the FFHQ dataset. For other datasets like LSUN Church, you need to first train the generator on this dataset, and then train the encoder. Finally, you need to extract the editing directions (this might be tricky; for FFHQ we can use the supervised label from the CelebA dataset, while for LSUN Church, you may need to rely on some unsupervised method). Putting all these together, you might be able to make it work on the LSUN Church dataset.

**Comment by MDGoncalves** [comment]
Hello! First of all thank you so much for your contribution. I'm trying to apply this code to a different Dataset as well. You mentioned that for FFHQ you used the supervised label from the CelebA dataset, but could you tell us what did you used for the LSUN Car dataset?

**Comment by tonylins** [comment]
Hi, for the LSUN Car dataset, you may refer to some unsupervised methods for discovering editing directions like https://arxiv.org/abs/2004.02546.

**Comment by MDGoncalves** [comment]
Thank you very much for the reply! Will give it a try
